handlers/libhoney_event_handler.go

```diff
@@ -86,40 +86,53 @@ func initLibhoney(config config.Config, version string) func() {
 	return libhoney.Close
 }
 
+// setTimestampsAndDurationIfValid sets time-related fields in the emitted telemetry
+// about the request/response cycle.
+//
+// It only sets timestamps if they are present in the captured event, and only
+// computes and includes durations for which there are correct timestamps to based them upon.
+func setTimestampsAndDurationIfValid(honeyEvent *libhoney.Event, httpEvent assemblers.HttpEvent) {
+	honeyEvent.AddField("meta.httpEvent_handled_at", time.Now())
+	switch true {
+	case httpEvent.RequestTimestamp.IsZero() && httpEvent.ResponseTimestamp.IsZero():
+		// no request or response, which is weird, but let's send what we do know
+		honeyEvent.AddField("meta.timestamps_missing", "request, response")
+		honeyEvent.Timestamp = time.Now()
+
+	case httpEvent.RequestTimestamp.IsZero():
+		// no request
+		honeyEvent.AddField("meta.timestamps_missing", "request")
+		// but we have a response
+		honeyEvent.Timestamp = httpEvent.ResponseTimestamp
+		honeyEvent.AddField("http.response.timestamp", httpEvent.ResponseTimestamp)
+		honeyEvent.AddField("meta.httpEvent_response_handled_latency_ms", time.Since(httpEvent.ResponseTimestamp).Milliseconds())
```

Comment:
```suggestion
		honeyEvent.AddField("meta.response.capture_to_handle.latency_ms", time.Since(httpEvent.ResponseTimestamp).Milliseconds())
```

![I don't know which.](https://media.tenor.com/-tb1sJkusiwAAAAM/not-sure.gif)

Reply:
I like these better than having httpEvent in the name.

Reply:
Agreed! Updating now to these new fields... and we can rename [again] in the future if needed 😅